### ISSUE TITLE
Grid name incompatibilty LPJ-GUESS - OIFS 48r1

### ISSUE THREAD

**Issue by hajlaci** [esm-tools/esm_tools#1257]
I'm trying to run the awiesm3 v3.4, which now has LPJ-GUESS coupled to it. It compiles without a problem; however, LPJ-GUESS requires that the grid size be set during compilation.
It's also hardcoded and only accepts three values namely T255, T159, or TL255.
As far as I know, the resolution for the atmosphere has different syntax (names) for these resolutions.
Can you help me with a workaround solution to run the coupled awiesm3?
 
I'm not sure if I explained it (or even understood it correctly), but I included Jan as well, so if something is not clear, he will be able to explain it in more detail.
 
The branch on which I'm working is dev/integrate-with-ece4 and all changes are committed and pushed to the branch and the relevant files are (I think):
.../esm_tools/configs/components/lpj_guess/lpj_guess.yaml
.../esm_tools/configs/setups/awiesm3/awiesm3.yaml

**Comment by JanStreffing** [comment]
Yeah so the issue is, that in the coupled awi-esm3, there are now three variables:

for esm_runscripts:
```
oifs:
  resolution:
lpj_guess
  resolution:
```
and for esm_master:
```
lpj_guess:
  mygrid:
```

The different nameing is actually not the issue here. Eventually the nameing for mygrid and resolution will be the same. The issue is, that we have three different variables, and the system only works when they are the same value. This brings lots of chances for user error. Especially mygrid, which can only be changed from withing the config/components/lpj_guess/lpj_guess.yaml.

**Comment by mandresm** [comment]
Hi @JanStreffing and @hajlaci, is this still a problem? I don't think I am understanding the issue correctly, but let me try:

LPJ-GUESS needs to be told which grid to use during compilation and that's in essence the whole problem correct? If someone uses a grid for compilation and runs with another grid they'll have problems. I see four possibilities:

Option 1: have a version of awiesm3 per grid for when you compile so that you run `esm_master install awiesm3-v3.4-T159`

Option 2: have one single version of awiesm3 in esm_tools but give esm_master a yaml file as input including the grid info using `esm_master install ... --modify-config config.yaml` (with `lpj_guess.resolution: T159`)

Option 3: compile lpj_guess once with every grid during esm_master, name each binary differently and then make the binary to be used for esm_runscripts depend on the resolution

Option 4: change LPJ_GUESS so that it supports different grids

Please, let me know if any of these options works for you so that I can expand on it.
